custom fbcode serialize stage to run FVP internally on arm ops tests

3l1 opened the pull request:
Summary:
update the codebase to allow a system whereby:

- the test inventory in https://www.internalfb.com/code/fbsource/fbcode/executorch/backends/arm/test/ops/, which until now runs on open source CI can all be exercised internally on our internal diff-time sandcastle and CI runs
- those tests can **transparently** switch between running an internal version of FVP using Meta's FVP runtimes, build configs and artifacts when run in fbsrouce, but then use the open-source FVp runtime when executed in github opens-ource CI

Differential Revision: D81159036

digantdesai reviewed (approved):
**Thank @3l1!**
LGTM, stamping to unblock but need some small fixes. And make sure the CI is clean.

3l1 committed:
custom fbcode serialize stage to run FVP internally on arm ops tests (#14070)

Summary:

update the codebase to allow a system whereby:

- the test inventory in https://www.internalfb.com/code/fbsource/fbcode/executorch/backends/arm/test/ops/, which until now runs on open source CI can all be exercised internally on our internal diff-time sandcastle and CI runs
- those tests can **transparently** switch between running an internal version of FVP using Meta's FVP runtimes, build configs and artifacts when run in fbsrouce, but then use the open-source FVp runtime when executed in github opens-ource CI

bypass-github-export-checks
bypass-github-pytorch-ci-checks
bypass-github-executorch-ci-checks

Reviewed By: digantdesai

Differential Revision: D81159036
